### DISCUSSION TITLE
[Feature Request] Easy way to add a noindex header for a production deployment (useful for project migrations)

### DISCUSSION THREAD

**Discussion by cbdeveloper** [vercel/community#1061] [Help] [answered]
I'm currently migrating a project to Vercel. But I would like to test my production deployment for a few days, before connecting my custom domain to it and allowing robots to crawl it.

The issue is that Vercel allows production deployments to be crawled and indexed by default, which makes sense for day-to-day use, but during a migration of a project can be tricky to deal with.

Ideally we would have a deployment settings that would allow us to enable/disable the `x-robots-tag: noindex` header on a production deployment. Maybe this config would be suitable for the DANGER ZONE, since production deployments usually should be indexed.

I tried using `vercel.json` to conditionally add the header to `host: vercel.app` but it's not working.
In my case, it would be enough to add the header to the `vercel.app` domain, because I still didn't connect my custom domain to it.

**vercel.json**

```
"headers": [
    {
      "source": "/:path*",
      "has": [
        {
          "type": "host",
          "value": "vercel.app"
        }
      ],
      "headers": [
        {
          "key": "x-robots-tag",
          "value": "noindex"
        }
      ]
    },
]
```

I also tried with the following, without success:

```
      "has": [
        {
          "type": "host",
          "value": "*.vercel.app"
        }
      ],
```

Let me know what you think.

**Comment by amyegan** [comment, marked as answer]
Hi @cbdeveloper! I shared a link about setting a `x-robots-tag` header in your other post (https://github.com/vercel/community/discussions/1060#discussioncomment-4158708), but it looks like you already found that solution on your own. 😁 

When using `vercel.json` to set headers, the host value needs to match your site address. So it should be your site's subdomain, like `my-example-site.vercel.app`, instead of just `vercel.app`. If you make that modification, this config should work for you. Please let me know if you run into any more trouble with this!

**Reply by cbdeveloper** [reply]
Hey Amy,

Thanks for you reply!

I ended up not going with the custom `headers` solution for my issue.

I have multiple Vercel projects connected to that same repo, because I deploy the same app for multiple countries.
Because of that, I would need to keep track of multiple subdomains/domains in the `headers` config. I think that would be weird and hard to maintain.

Since my solution of adding the header to any `*.vercel.app` domain wasn't possible, this is what I decided to do in order to disallow robots from my production website deployment during the migration. Note: this is before connecting the custom domain and actually completing the migration.

I create the `robots.txt` file on the fly using an `/api` route.

In that route, I fetch a document from my DB that contains an `allow: boolean` flag.

If `allow is false`, I return a `robots.txt` file that won't allow robots to crawl my website.

Something like:

```
User-agent: *
Disallow: /
```

When I finish the website migration and connect the custom domain to it, I set `allow: true` in my DB and things go back to normal.

**Reply by amyegan** [reply]
Nice! Thanks for sharing this solution
